### ISSUE TITLE
Mixin error on SpongeForge server

### ISSUE THREAD

**Issue by Mr-Krab** [TerminatorNL/Tiquality#24]
Minecraft: 1.12.2
SpongeAPI: 7.1.0-b2736425
Sponge: 1.12.2-7.1.6-SNAPSHOT
SpongeForge: 1.12.2-2768-7.1.6-RC3641
Minecraft Forge: 14.23.5.2796

Mods (56): Minecraft, Minecraft Coder Pack, Forge Mod Loader, Minecraft Forge, SpongeAPI, Sponge, AI Improvements, Applied Energistics 2, Better Questing, BiblioCraft, Binnie Core, Binnie's Botany, Binnie's Design, Binnie's Extra Bees, Binnie's Extra Trees, Binnie's Genetics, CoFH Core, CoFH World, CodeChicken Lib, Console Filter, Construct's Armory, CraftTweaker JEI Support, CraftTweaker2, CreativeCore, CreativeCoreDummy, CustomNPCs, Ex Nihilo Creatio, Fishing Net Mod, Forestry, HoloInventory, Iron Chest, ItemPhysic, Just Enough Items, MTLib, Mantle, McJtyLib, Mekanism, Mod Tweaker, PortalGun, Quick Leaf Decay, Redstone Arsenal, Redstone Flux, Shadowfacts' Forgelin, SpongeForge, Standard Expansion, Thermal Expansion, Thermal Foundation, Thermal Innovation, Thermal Tinkering, Tinkers' Construct, Tiquality, Tree Chopper, UniDict, WanionLib, XNet, iChunUtil

Plugins (26): Minecraft, Minecraft Coder Pack, SpongeAPI, Sponge,  FastAsyncWorldEdit, BeanCore, CatClearLag, CommandSync, DupeFixer, EconomyLite, GriefPrevention, GuiShopManager, Holograms, HuskyCrates, HuskyUI, LuckPerms, Nucleus, ProjectCore, SkyClaims, StackBan, UltimateChat, UniversalMarket, VirtualTool, Wasted, WorldEdit, spark

After the server has finished loading, an error appears. -> https://pastebin.com/X3MiF7Mp

**Comment by TerminatorNL** [comment]
SpongeForge restructured their internals effectively invalidating tiquality's Mixin instructions. You can attempt to downgrade SpongeForge or wait for a next Tiquality release. I will update accordingly.

**Comment by Mr-Krab** [comment]
I'll wait. I also ask you to recycle the import regions of the GriefPrevention plugin. It quickly overloads the server and it can cause crash. It is worth making pauses during this process.

The Sponge configuration has the ability to make global tile slowdowns more efficient. If possible, add to your config an option when enabled, which your mod will only slow down the player’s tiles if it has exceeded its server load limit. This will reduce the load from your mod to the main server thread.

**Comment by TerminatorNL** [comment]
It's worth noting that the GriefPrevention import command is not an optimal solution, as it has to load all the chunks it needs to save data to.
I don't really understand what you mean with 'recycling', if you mean deleting that feature I probably won't because the owner of the server can decide it might be useful for them. Most times the import fails is due to mods having corrupt chunks, and Tiquality attempts to load these.

As for your second question, the SpongeForge slowdown configuration is out of Tiquality's scope for now. I need to make Tiquality as stable as I can before moving into specifics.

Please note that throttling done by SpongeForge **should** not be needed when Tiquality is configured correctly. (Should being the key-word here.) as Tiquality should fix the lag introduced by the tile entities essentially the same way SpongeForge does (Running the update less frequently).

If you know what blocks you want to throttle, the best bet would be to go to the respective mod author and kindly ask them to optimize their code. Not all mod authors will like feedback like this, the best you can do is include reports from several profilers. Profilers help them out and is a good way of showing them that you care, and are willing to help instead of just throwing problems at them and then moving on.

In the meanwhile, I have submitted a report at https://github.com/SpongePowered/SpongeForge/issues/2678 asking for help about the main issue in this thread. ;)

**Comment by Mr-Krab** [comment]
I made such requests to mod developers. For example, the CoFH team said that the load created by their Thermal Dinamics mod is quite normal. Although in fact she is huge and this mod prevents chunks from being normally unloaded. As a result, I got rid of this mod and took xNet, but its localization is not complete, I can not translate it into Russian completely, I have to compile it myself.
Of course, there are adequate mod developers, such as you or the authors of Mekanism, but in most cases the situation is very sad.
